File: src/dotnet/Ansight/Pairing/Models/DeviceNetworkProfile.cs
namespace Ansight.Pairing.Models;

public sealed class DeviceNetworkProfile
{
    public int? TransportCode { get; set; }
    public bool? Metered { get; set; }
    public string? EffectiveType { get; set; }
    public int? RttMs { get; set; }
    public int? DownKbps { get; set; }
}

File: src/ios/Sources/AnsightKit/DeviceNetworkProfile.swift
import Foundation

#if canImport(Darwin)
import Darwin
#endif

#if canImport(UIKit)
import UIKit
#endif

public struct DeviceNetworkProfile: Sendable, Codable, Equatable {
    public var transportCode: Int?
    public var metered: Bool?
    public var effectiveType: String?
    public var rttMs: Int?
    public var downKbps: Int?
}
